### PR TITLE
Add check for profiling results.

### PR DESCRIPTION
**Pull request by gartung**

### CODE REVIEW

**Comment by gartung** [comment]
requires https://github.com/cms-sw/cmssdt-ib/pull/15

**Comment by smuzaffar** [comment]
@gartung , I think you also need to add profile entry like https://github.com/cms-sw/cms-bot/blob/1cd86d24e852f12c846951e6997a328c5e84fcbe/report-summary-merged-prs#L848

File: report-summary-merged-prs [hunk 1]
```diff
@@ -976,6 +979,23 @@ def find_dup_dict_result(comparisons):
             })
 
 
+def find_profiling_results(test_field, comparisons, architecture, magic_command, results_function=find_one_test_results):
```
Comment (smuzaffar) [inline report-summary-merged-prs:982]: no need to add this fucntion

Comment (gartung) [reply report-summary-merged-prs:982]: OK. Let me try this using the default function. There are two files that need to be found before the links are created.

File: report-summary-merged-prs [hunk 2]
```diff
@@ -1404,6 +1423,8 @@ if __name__ == "__main__":
     MAGIC_COMMAND_FIND_MATERIL_BUDGET_COMPARISON_CHECKS = "TEST_FILE=" + JENKINS_ARTIFACTS_DIR + "/material-budget/RELEASE_NAME/ARCHITECTURE/comparison/Images/MBDiff.txt && test -f $TEST_FILE && grep '0$' $TEST_FILE | wc -l"
     MAGIC_COMMAND_FIND_VALGRIND = 'test -d ' + JENKINS_ARTIFACTS_DIR + '/valgrind/RELEASE_NAME'
     MAGIC_COMMAND_FIND_IGPROF = 'test -d ' + JENKINS_ARTIFACTS_DIR + '/igprof/RELEASE_NAME'
+    MAGIC_COMMAND_FIND_PROFILING_CHECKS_FILTER1 = 'test -f ' + JENKINS_ARTIFACTS_DIR + '/profiling/RELEASE_NAME/ARCHITECTURE/23434.21/step3_RAW2DIGI_L1Reco_RECO_RECOSIM_PU.resources.json'
+    MAGIC_COMMAND_FIND_PROFILING_CHECKS_FILTER2 = 'test -f ' + JENKINS_ARTIFACTS_DIR + '/igprof/RELEASE_NAME/ARCHITECTURE/profiling/23434.21/sorted_RES_CPU_step3.txt'
```
Comment (smuzaffar) [inline report-summary-merged-prs:1427]: just echo the file name to be saved in the json file e.g. see this https://github.com/cms-sw/cms-bot/blob/master/report-summary-merged-prs#L1402 . In your case you can echo "ARCHITECTURE/profiling/23434.21/sorted_RES_CPU_step3.txt/sorted_RES_CPU_step3.txt" and create a simple function to either see the result to "inprogress" or {"status": "passed", "data": "yourfile"}

you can then pass this function to find_general_test_results 

I can update this if this looks complicated to you

File: report-summary-merged-prs (hunk 1 above)
Comment (gartung) [reply report-summary-merged-prs:982]: But find_general_test_results do not add the architecture field.

Comment (smuzaffar) [reply report-summary-merged-prs:982]: just set  results_function=your_callback_fucntion for https://github.com/cms-sw/cms-bot/blob/master/report-summary-merged-prs#L979 and then your your_callback_fucntion can set architecture and workflow number . Avoid hard coding the workflow too.

**Comment by gartung** [comment]
This PR is producing the output I expect now
https://cmssdt.cern.ch/jenkins/job/test-summary-merged-prs/

**Pull request merged**